inference-engine/src/transformations/src/transformations/common_optimizations/broadcast_elementwise_fusion.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/broadcast_elementwise_fusion.hpp"
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::BroadcastElementwiseFusion, "BroadcastElementwiseFusion", 0);
+
+ngraph::pass::BroadcastElementwiseFusion::BroadcastElementwiseFusion() {
+    auto input1 = ngraph::pattern::any_input();
+    auto input2 = ngraph::pattern::any_input();
+    auto broadcast = ngraph::pattern::wrap_type<ngraph::opset5::Broadcast>({input2, pattern::any_input()});
+    auto elementwise = ngraph::pattern::wrap_type<ngraph::op::util::BinaryElementwiseArithmetic>({input1, broadcast});
```

Comment:
What if the `broadcast` is the first input and `input1` is the second input for the Eltwise? Is this case we still can apply the transformation but your transformation will not work.

Reply:
Why is this marked as resolved because I still see `broadcast` to be the second input?

Reply:
The question is still not covered.

Reply:
We have EliminateBroadcastSwapInputsTest test case which covered that case. So it was resolved.